packages/recommend-react/src/useLookingSimilar.ts

```diff
@@ -29,7 +29,9 @@ export function useLookingSimilar<TObject>({
   const queryParameters = useStableValue(userQueryParameters);
   const fallbackParameters = useStableValue(userFallbackParameters);
 
-  const { hasProvider, register } = useRecommendContext();
+  const { hasProvider, register } = useRecommendContext<
+    GetRecommendationsResult<TObject>
```

Comment:
Couldn't this generic be passed to the `register()` call?

This line sounds like a lie. The Recommend context isn't of this type, you want to register a new model with this type.

We can further improve in follow up PRs.

Reply:
That was my initial attempt, as it indeed makes more sense!

However, getting the generic type to "propagate" from `register` to the rest of the Recommend context wasn't super obvious after.